src/utils.py

```diff
@@ -15,27 +15,27 @@
 api_url = os.environ.get("HARBOR_API_URL")
 
 
-def check_file_exists(path: str) -> bool:
+def check_file_exists(path: str, logger) -> bool:
```

Comment:
Could we maybe change the name from `check_file_exists` to `file_exists`?

It sounds more natural to write `if file_exists(path)`